src/back/spv/block.rs

```diff
@@ -1120,6 +1141,101 @@ impl<'w> BlockContext<'w> {
         Ok(pointer)
     }
 
+    /// Build the instructions for the arithmetic expression of a dot product
+    fn write_dot_product(
+        &mut self,
+        result_id: Word,
+        result_type_id: Word,
+        arg0_id: Word,
+        arg1_id: Word,
+        size: crate::VectorSize,
+        block: &mut Block,
+    ) -> Result<(), Error> {
+        // Create an iterator over the IDs that will store the partial sums of products.
+        // Last iterator element will always be result_id, while the others will be generated IDs.
+        let sum_2_ids: [Word; 1];
+        let sum_3_ids: [Word; 2];
+        let sum_4_ids: [Word; 3];
+        use crate::VectorSize;
+        let sum_ids = match size {
```

Comment:
I think we can simplify this by starting off `OpConstantNull` of type `result_type_id` and just adding to it in a loop